openff/toolkit/topology/molecule.py

```diff
@@ -2726,6 +2735,73 @@ def __eq__(self, other):
         # TODO the doc string did not match the previous function what matching should this method do?
         return Molecule.are_isomorphic(self, other, return_atom_map=False)[0]
 
+    def _add_default_hierarchy_schemes(self):
+        self.add_hierarchy_scheme(
+            ("chain", "residue_number", "residue_name"), "residues"
+        )
+        self.add_hierarchy_scheme(("chain",), "chains")
+
+    def add_hierarchy_scheme(
+        self,
+        uniqueness_criteria,
+        iterator_name,
+    ):
+        """
+        Parameters
+        ----------
+        uniqueness_criteria : tuple of str
+        iterator_name : str
+            Name of the iterator that will be exposed to access the HierarchyElements generated
+            by this scheme
+        sort_func : function
+            The function for deciding the iteration order of HierarchyElements generated by this scheme.
+
+        Returns
+        -------
+        new_hier_scheme : openff.toolkit.topology.HierarchyScheme
+            The newly created HierarchyScheme
+        """
+        assert iterator_name not in self._hierarchy_schemes
```

Comment:
```suggestion
        if iterator_name in self._hierarchy_schemes:
            raise IteratorNameAlreadyInExistingHierarchySchemesOrSomethingOrOtherError
```

I ran into this while calling `._add_default_hierarchy_schemes()` a second time on the same molecule, so I think it will be a relatively common exception to hit and worth ma king a custom exception

Reply:
Fixed in fba36aa